runtime/runtime/src/tests/apply.rs

```diff
@@ -2793,3 +2837,186 @@ fn test_deploy_and_call_local_receipts() {
+        alice_tx1.get_hash(),
+        bob_tx2.get_hash(),
+        alice_tx2.get_hash(),
+        bob_tx3.get_hash(),
+    ];
+
+    // Note: The 3 local receipts are generated for valid transactions
+    // where signer_id == receiver_id - tx2, tx4, tx6 (not for tx1 as it is dropped).
+    assert_eq!(
+        apply_result.outcomes.len(),
+        8,
+        "should have processed 5 transactions and 3 local receipts"
+    );
+    let tx_outcomes = apply_result.outcomes.iter().take(5).map(|o| o.id).collect::<Vec<_>>();
+    assert_eq!(tx_outcomes, expected_order, "outcomes are not in expected sorted order");
+}
+
+/// Verifies proper ordering and balance update for transactions signed with multiple keys from one account.
+/// Alice is set up with 3 full-access keys.
+/// Six transactions from Alice to Bob are submitted using various nonces and keys.
+/// The test checks that outcomes are correctly ordered and Alice's final balance is within the expected range.
+#[test]
+fn test_transaction_multiple_access_keys_with_apply() {
+    let alice_signer1 = InMemorySigner::from_seed(alice_account(), KeyType::ED25519, "seed1");
+    let alice_signer2 = InMemorySigner::from_seed(alice_account(), KeyType::ED25519, "seed2");
+    let alice_signer3 = InMemorySigner::from_seed(alice_account(), KeyType::ED25519, "seed3");
+
+    let send_money_tx = |nonce, key| {
+        SignedTransaction::send_money(
+            nonce,
+            alice_account(),
+            bob_account(),
+            key,
+            to_yocto(1000),
+            CryptoHash::default(),
+        )
+    };
+
+    let txs = vec![
+        send_money_tx(1, &alice_signer1),
+        send_money_tx(1, &alice_signer2),
+        send_money_tx(1, &alice_signer3),
+        send_money_tx(2, &alice_signer3),
+        send_money_tx(2, &alice_signer1),
+        send_money_tx(3, &alice_signer1),
+    ];
+
+    let accounts_with_keys = vec![
+        (
+            alice_account(),
+            vec![Arc::new(alice_signer1), Arc::new(alice_signer2), Arc::new(alice_signer3)],
+        ),
+        (bob_account(), vec![]),
+    ];
+
+    let (runtime, tries, root, mut apply_state, _signers, epoch_info_provider) =
+        setup_runtime_with_keys(
+            accounts_with_keys,
+            to_yocto(1_000_000),
+            to_yocto(500_000),
+            10u64.pow(15),
+        );
+
+    let flags = vec![true; txs.len()];
+    let s_valid_txs = SignedValidPeriodTransactions::new(&txs, flags.as_slice());
+    let apply_result = runtime
+        .apply(
+            tries.get_trie_for_shard(ShardUId::single_shard(), root),
+            &None,
+            &apply_state,
+            &[],
+            s_valid_txs,
+            &epoch_info_provider,
+            Default::default(),
+        )
+        .expect("apply should succeed");
+
+    let expected_order = txs.iter().map(|tx| tx.get_hash()).collect::<Vec<_>>();
+
+    assert_eq!(apply_result.outcomes.len(), 6, "should have processed 6 transactions");
```

Comment:
```suggestion
    assert_eq!(apply_result.outcomes.len(), txs.len(), "should have processed 6 transactions");
```

Isn't it tx.len()?  Why hardcode 6?